plugins/processors/t128_filter/t128_filter.go

```diff
@@ -280,15 +280,20 @@ func compileGlobs(values []string) ([]glob.Glob, error) {
 }
 
 func (r *T128Filter) Apply(in ...telegraf.Metric) []telegraf.Metric {
-	filteredPoints := make([]telegraf.Metric, 0)
-
 	for _, point := range in {
-		if r.matcher.Matches(point) {
-			filteredPoints = append(filteredPoints, point)
+		if !r.matcher.Matches(point) {
+			// copying so that don't hit seg fault
+			fields := make([]*telegraf.Field, len(point.FieldList()))
+			copy(fields, point.FieldList())
+
+			// removing all fields will have telegraf drop the metric
+			for _, field := range fields {
```

Comment:
They do have this method:
```go
	// Drop marks the metric as processed successfully without being written
	// to any output.
	Drop()
```
which on the surface looks like what we want. But I believe it actually deals with verifying a metric was successfully written to and output (or at least handled by it). It marks the metric as being successfully handled via a drop, but I don't believe it causes the metric to be dropped.

Reply:
Yea, I looked into that and tried it on a real system and it did not drop the metric in this scenario. I also wish there was an easier way to drop all of the fields, but this was the only way that I found. The copy of the slice was necessary because I would get a seg fault when iterating while calling `RemoveField` on each key.